staging/src/k8s.io/apiserver/pkg/cel/library/network_addrs.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"net/netip"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+
+	apiservercel "k8s.io/apiserver/pkg/cel"
+)
+
+// NetworkAddr provides a CEL function library extension of network address parsing functions.
+//
+// networkAddr
+//
+// Converts a string to a network address or results in an error if the string is not a valid network address.
+// The network address must be an IPv4 or IPv6 address with a CIDR mask.
+//
+//	networkAddr(<string>) <NetworkAddr>
+//
+// Examples:
+//
+//	networkAddr('192.168.0.0/16') // returns an IPv4 address with a CIDR mask
+//	networkAddr('::1/128') // returns an IPv6 address with a CIDR mask
+//	networkAddr('192.168.0.0/33') // error
+//	networkAddr('::1/129') // error
```

Comment:
I'd include

```
     networkAddr('192.168.0.1/16') // error, because there are non-0 bits after the prefix
```

There are no APIs in `k8s.io` that _want_ an IP-with-prefix-length value rather than a subnet/mask value, but there are currently APIs that incorrectly allow them (https://github.com/kubernetes/kubernetes/pull/94484). We should forbid them here. If someone later needs an IP-with-prefix-length value (Multi-Network maybe?) they can add new functions for that.

Reply:
That is to say that, when we parse a CIDR in this string using `netip.ParsePrefix`, we want to also make sure that the result `Addr()` is the same as `Masked().Addr()` right? Masking the address we got it should not change the address

Reply:
right